### PR TITLE
Adds OS specific details to the user-agent header

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Adds OS-specific details to the user-agent header

# Testing

Screen Shot 2022-06-30 at 1 44 55 PM

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by Justintime50 — CHANGES_REQUESTED** [review]
The code looks fine, I would however like an `Implementation` piece included like we did with Python. What kind of Java are they using.

**Comment by jchen293** [comment]
> The code looks fine, I would however like an `Implementation` piece included like we did with Python. What kind of Java are they using.

added

**Comment by jchen293** [comment]
> 

This also applies to other sections as well, we need to add this in other libs or the one(s) that already rolled out.

**Comment by jchen293** [comment]
> So the way that our User-Agent header works is that each new "section" is delimited by a space. The `Implementation` section has spaces in it. We may want to join each piece of that section with an underscore or something similar so that the sections are consistent. Does that make sense? (this is based on the new pic)

> 

This also applies to other sections as well, we need to add this in other libs or the one(s) that already rolled out.

**Comment by Justintime50** [comment]
> This also applies to other sections as well, we need to add this in other libs or the one(s) that already rolled out.

Can you supply an example of this happening? I don't believe this is actually the case. In Python, the `Implementation` will be something like `Pypy` or `CPython` and won't have extra spaces in it.

**Comment by jchen293** [comment]
> > This also applies to other sections as well, we need to add this in other libs or the one(s) that already rolled out.
> 
> Can you supply an example of this happening? I don't believe this is actually the case. In Python, the `Implementation` will be something like `Pypy` or `CPython` and won't have extra spaces in it.

Ok, in Python this probably will not be an issue, but for the `OS name`, in my case, it would be `Mac OS X`, we definitely want to convert spaces to hyphens.

File: src/test/java/com/easypost/TestUtils.java [hunk 1]
```diff
@@ -18,12 +18,11 @@ public enum ApiKey {
         PRODUCTION
     }
 
-    private static final String API_KEY_FAILED_TO_PULL = "couldnotpullapikey";
+    private static final String API_KEY_FAILED_TO_PULL = "couldNotPullApiKey";
```
Comment (nwithan8) [inline src/test/java/com/easypost/TestUtils.java:21]: Why the change?

**Review by nwithan8 — APPROVED** [review]

File: src/test/java/com/easypost/TestUtils.java (hunk 1 above)
Comment (jchen293) [reply src/test/java/com/easypost/TestUtils.java:21]: I prefer the camel case in Java

**Review by Justintime50 — APPROVED** [review]
